src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -256,7 +255,12 @@ protected function resolveExprConstFetch(Expr\ConstFetch $node)
 
     protected function resolveExprClassConstFetch(Expr\ClassConstFetch $node)
     {
-        $refClass     = $this->fetchReflectionClass($node->class);
+        $classToReflect = $node->class;
```

Comment:
Yes, I know you can do a constant fetch from a class name stored in a variable. I'm unsure what other `Expr`s are valid. I believe *ParserReflection* uses this mostly for evaluating constants, so this might not be a valid code path as we are using it, but it is a valid value for `$node->class` in some circumstances.

As far as adding corresponding test cases, I agree, but I was lumping my test changes into a separate PR. (I'm shooting for 100% code coverage, but I've always been a cockeyed optimist.) I can add these tests to this PR after I've untangled my current repo branches. (I don't really want to do a stash right now.)